crates/ruff/src/rules/flake8_pyi/rules/type_comment_in_stub.rs

```diff
@@ -0,0 +1,59 @@
+use rustpython_parser::lexer::LexResult;
+use rustpython_parser::Tok;
+
+use ruff_macros::{define_violation, derive_message_formats};
+
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+use crate::Range;
+
+define_violation!(
+    /// ## What it does
+    /// Do not use type comments (e.g. `x = 1 # type: int`) in stubs, even if
+    /// the stub supports Python 2. Always use annotations instead (e.g.
+    /// `x: int = 1`).
+    ///
+    /// ## Why is this bad?
+    /// You should use type annotation directly in pyi files.
+    ///
+    /// ## Example
+    /// ```python
+    /// x = 1 # type: int
+    /// ```
+    ///
+    /// Use instead:
+    /// ```python
+    /// x: int = 1
+    /// ```
+    pub struct TypeCommentInStub;
+);
+impl Violation for TypeCommentInStub {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Don't use type comments in stub file")
+    }
+}
+
+/// PYI033
+pub fn type_comment_in_stub(tokens: &[LexResult]) -> Vec<Diagnostic> {
+    let mut diagnostics = vec![];
+
+    for token in tokens.iter().flatten() {
+        if let (location, Tok::Comment(comment), end_location) = token {
+            // I couldn't find any PEP on the exact syntax (the closest being
+            // https://peps.python.org/pep-0484/#type-comments), but every case I saw used
+            // `# type:` verbatim so this seems to be the right thing to pick
+            if comment.starts_with("# type:") {
```

Comment:
Would this also be true for `# type: ignore`?

Reply:
currently yes; are there any cases where you need `# type: ignore` in a .pyi file?

Reply:
I wish ignore comments wouldn't be needed in stubs but they are used. Here is an example from typeshed https://github.com/python/typeshed/blob/45f0a5e7e47a23d6be67cfce429c4f2c0de62a29/stdlib/builtins.pyi#L565

Reply:
thanks for the example! i'll except those then

Reply:
All our flake8-pyi tests for this rule are in this file FYI, if that helps :)

https://github.com/PyCQA/flake8-pyi/blob/main/tests/type_comments.pyi

Reply:
It might be worth looking at the `flake8-pyi` source and seeing how they handle this?

In particular, they have this regex, which omits `ignore` comments:

https://github.com/PyCQA/flake8-pyi/blob/66f28a4407af2156f95d370941fae4dd98280740/pyi.py#L1877

Reply:
oops thanks i somehow used them for Y006 but then forgot that those also exist for Y033.

thanks!

Reply:
fwiw i had to do two regexes since rust regex doesn't support look around

Reply:
Thanks @AlexWaygood for the helpful comment :)